feature: 기본 엔티티 생성

seokhyeong9171 opened the pull request:
### 변경사항
<!-- 이 PR에서 어떤점들이 변경되었는지 기술해주세요. 가급적이면 as-is, to-be를 활용해서 작성해주세요.  -->
**AS-IS**

**TO-BE**

domain 경로에 erd 참고하여 기본 엔티티를 생성하였습니다.

### 테스트
<!-- 본 변경사항이 테스트가 되었는지 기술해주세요 --> 
- [ ] 테스트 코드
- [ ] API 테스트

seokhyeong9171 committed:
feature: 기본 엔티티 생성

rrr7777777goo commented:
휴무일 테이블과 연동되는 Holiday 클래스를 추가해야 할 것 같습니다.

amj9843 commented on src/main/java/com/zerobase/backend/domain/ChatRoom.java:
```diff
@@ -0,0 +1,35 @@
+  @Column(name = "chat_room_id", nullable = false)
+  private Long id;
+
+  @OneToOne(fetch = FetchType.LAZY)
```
저희 이렇게 값이 중복될 수 없는 항목엔 unique 어노테이션으로 체크하는건 어떨까요? 로직상 하나만 들어갈수밖에 없다곤 하나 혹시나 하는 마음에..

amj9843 commented on src/main/java/com/zerobase/backend/domain/Entrepreneur.java:
```diff
@@ -0,0 +1,56 @@
+  private Long id;
+
+  @Column(nullable = false)
+  private String email;
```
마찬가지로 회원정보에선 이메일을 유니크로 설정하면 될 것 같습니다!

seokhyeong9171 committed:
refactor: 조원 피드백 반영
- ChatRoom의 meeting field에 unique 제약 추가
- User, Entrepreneur의 email에 unique 제약 추가
- Evaluate의 type 필드 삭제
- Evaluate의 EvaluateType을 EvaluateBadge로 변경
- Holiday class 추가
- Menu에 동일메뉴 관련 unique 제약 추가
